From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spottedleaf <Spottedleaf@users.noreply.github.com>
Date: Mon, 25 Nov 2019 05:45:45 -0800
Subject: [PATCH] Avoid immediately autosaving newly generated/loaded chunks

We shouldn't be saving them so soon after loading/generating, and
this specifically optimises elytra flying

diff --git a/src/main/java/net/minecraft/server/PlayerChunkMap.java b/src/main/java/net/minecraft/server/PlayerChunkMap.java
index 68636931a39cc6f6b24c135a33da3efcf8fa833e..97f93712e44235283395c491f89114af7f6b3307 100644
--- a/src/main/java/net/minecraft/server/PlayerChunkMap.java
+++ b/src/main/java/net/minecraft/server/PlayerChunkMap.java
@@ -991,6 +991,8 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
                     playerchunk.a(new ProtoChunkExtension(chunk));
                 }
 
+                chunk.setLastSaved(this.world.getTime()); /* Concrete - avoid autosaving newly generated/loaded chunks */
+
                 chunk.a(() -> {
                     return PlayerChunk.getChunkState(playerchunk.getTicketLevel());
                 });
